### ISSUE TITLE
qwen3.5 4b grpo训练，模型只读分析图片信息，忽略user prompt

### ISSUE THREAD

**Issue by yangqifanbq** [modelscope/ms-swift#8455]
### Checklist / 检查清单

- [x] I have searched existing issues, and this is a new question or discussion topic. / 我已经搜索过现有的 issues，确认这是一个新的问题与讨论。

### Question Description / 问题描述

ms_swift版本4.1.0.dev0

在qwen3.5 4b grpo训练上think输出太长了，导致答案提取不到，1.prompt添加回答样例样例，以及规定不超过200token2.改max_completion_length和训练参数3.换9b，都解决不了

修改了swift/template/templates/qwen.py中，is_thinking改成false，期望改成instruct，改动如下
<img width="1052" height="303" alt="Image" src="https://github.com/user-attachments/assets/fe977f26-4982-4d8b-b870-f12a8eaca74f" />
现在模型只读分析图片信息，忽略user prompt（从输出看，模型读取不到文本输入，如：由于我无法看到这些文字的具体内容，我无法确认它们是否准确或完整。建议您提供更多的上下文信息，以便我能够更好地帮助您）

**Comment by hjh0119** [comment]
```
	--enable_thinking false \
	--max_length xxx \(prompt_length for grpo)
```

**Comment by yangqifanbq** [comment]
> ```
> 	--enable_thinking false \
> 	--max_length xxx \(prompt_length for grpo)
> ```

我将is thinking重新修改为true，以及在bash中添加了-enable_thinking false ，模型都是对图像分析以及读取不到文本输入，模型输出如：
您好！我注意到您发送的内容似乎包含了一些重复的、非标准的文本信息。为了能更好地帮助您，我尝试忽略这些干扰项，专注于理解您可能想要传达的核心内容。不过，从这些文本的片段来看，它们似乎与常规的中文交流无关。因此，我无法确定您具体想要表达什么。



数据集格式为
{
  "messages": [
    {
      "role": "system",
      "content": "内容"
    },
    {
      "role": "user",
      "content": "<image>\n任务描述文本"
    }
  ],
  "images": [
    {
      "path": "图像文件路径"
    }
  ],
  "reward_model": {
    "ground_truth": "<answer> 答案 </answer>",
    "style": "model"
  },
  "answer_choice": "答案",
  "solution": "<answer> 答案 </answer>",
  "extra_info": {
    "answer": "<answer> 答案 </answer>",
    "index": "样本索引",
    "split": "数据集划分",
    "width": null,
    "height": null
  }
}

尝试查看ORM类奖励函数的message变量的sys prompt，user prompt输入，没有问题

**Comment by AriesJin** [comment]
你的grpo会很慢吗，并且显存不均衡，并且占用很大

**Comment by yangqifanbq** [comment]
> 你的grpo会很慢吗，并且显存不均衡，并且占用很大


{'loss': '0', 'grad_norm': '2', 'learning_rate': '1e-08', 'completions/mean_length': '708.5', 'completions/min_length': '91', 'completions/max_length': '2048', 'completions/clipped_ratio': '0.2344', 'reward': '0', 'reward_std': '0', 'frac_reward_zero_std': '1', 'rewards/MCPReward/mean': '0', 'rewards/MCPReward/std': '0', 'rewards/ThinkFormatReward/mean': '0', 'rewards/ThinkFormatReward/std': '0', 'kl': 'nan', 'clip_ratio/low_mean': '0', 'clip_ratio/low_min': '0', 'clip_ratio/high_mean': '0', 'clip_ratio/high_max': '0', 'clip_ratio/region_mean': '0', 'step_time': '322.9', 'epoch': '0.003636', 'global_step/max_steps': '1/1375', 'elapsed_time': '10m 47s', 'remaining_time': '10d 6h 46m 35s', 'memory(GiB)': '46.97', 'train_speed(s/it)': '646.6'}
在2张a6000上做测试，这是第一轮训练情况，截断部分均为模型复读情况，prompt为英文输出有时为中文

export PYTORCH_CUDA_ALLOC_CONF="max_split_size_mb:512,expandable_segments:True"
export NCCL_P2P_DISABLE=1
export NCCL_IB_DISABLE=1
export CUDA_LAUNCH_BLOCKING=1          # 调试模式，定位错误
export SWIFTLIB_USE_FLASH_ATTENTION=1  # Swift库级别启用
export MASTER_PORT=3010
export NPROC_PER_NODE=2
export OMP_NUM_THREADS=2

export CUDA_VISIBLE_DEVICES=0,1

 swift rlhf \
    --rlhf_type grpo \
    --model /mnt/nas/hf_ckpts/Qwen/Qwen3.5-4B \
    --model_type qwen3_5 \
    --external_plugins /home/qifan/resgo/resgo_s1.py\
    --reward_funcs mcp_stage1 think_format \
    --reward_weights 2 1 \
    --tuner_type full \
    --torch_dtype bfloat16 \
    --dataset /mnt/nas/qifan/train_s1_rlhf314_sys.jsonl \
    --max_completion_length 2048 \
    --num_train_epochs 5 \
    --per_device_train_batch_size 2 \
    --gradient_accumulation_steps 8 \
    --per_device_eval_batch_size 4 \
    --learning_rate 5e-7 \
    --beta 0.005 \
    --eval_steps 100 \
    --save_steps 40 \
    --save_total_limit 1 \
    --logging_steps 5 \
    --max_length 2048 \
    --output_dir /mnt/nas/qifan/resgo_s1 \
    --warmup_ratio 0.05 \
    --dataloader_num_workers 4 \
    --dataset_num_proc 4 \
    --num_generations 4 \
    --temperature 0.7 \
    --log_completions true \
    --async_generate false \
    --gradient_checkpointing true\
    --deepspeed zero3_offload \
    --enable_thinking false

**Comment by AriesJin** [comment]
4b模型，然后你的长度是2k，看着也很慢

**Comment by hjh0119** [comment]
1. `max_length=2048` is too small—it needs to account for visual token length as well.
2. Set some environment variables for image processing. Refer to: https://swift.readthedocs.io/zh-cn/latest/Instruction/Command-line-parameters.html#id19
3. The default inference sampling process is slow. We recommend using vLLM to accelerate inference.
